src/components/skills-quiz-v2/JobCardComponent.jsx

```diff
@@ -2,42 +2,76 @@ import React, { useContext, useState, useEffect } from 'react';
 import {
   SelectableBox, Chip, Spinner, Stack, Button,
 } from '@edx/paragon';
+import { camelCaseObject } from '@edx/frontend-platform/utils';
+import { logError } from '@edx/frontend-platform/logging';
 import PropTypes from 'prop-types';
 import { SkillsContext } from '../skills-quiz/SkillsContextProvider';
 import { SET_KEY_VALUE } from '../skills-quiz/data/constants';
-import { DROPDOWN_OPTION_IMPROVE_CURRENT_ROLE } from '../skills-quiz/constants';
 import TopSkillsOverview from '../skills-quiz/TopSkillsOverview';
 import SearchCourseCard from '../skills-quiz/SearchCourseCard';
 import SearchProgramCard from '../skills-quiz/SearchProgramCard';
 import SearchPathways from '../skills-quiz/SearchPathways';
 import SkillsCourses from '../skills-quiz/SkillsCourses';
+import { fetchCourseEnrollments } from '../skills-quiz/data/service';
+import { saveSkillsGoalsAndJobsUserSelected } from '../skills-quiz/data/utils';
 
 const JobCardComponent = ({
   jobs, isLoading, jobIndex, courseIndex,
 }) => {
   const { dispatch, state } = useContext(SkillsContext);
-  const { goal } = state;
+  const { goal, currentJobRole, interestedJobs } = state;
   const [jobSelected, setJobSelected] = useState(undefined);
   const [showMoreRecommendedCourses, setShowMoreRecommendedCourses] = useState(false);
 
   useEffect(() => {
-    if (jobs?.length === 1) {
+    if (jobs?.length > 0) {
       setJobSelected(jobs[0]?.name);
-      dispatch({ type: SET_KEY_VALUE, key: 'selectedJob', value: jobSelected });
-    } else if (jobs?.length === 0) {
-      setJobSelected(undefined);
-      dispatch({ type: SET_KEY_VALUE, key: 'selectedJob', value: undefined });
+      dispatch({
+        type: SET_KEY_VALUE,
+        key: 'selectedJob',
+        value: jobs[0]?.name,
+      });
     }
-  }, [jobs, dispatch, jobSelected]);
+  }, [jobs, dispatch]);
+
+  useEffect(() => {
+    if (goal && (currentJobRole || interestedJobs)) {
+      saveSkillsGoalsAndJobsUserSelected(goal, currentJobRole, interestedJobs);
+    }
+  }, [goal, currentJobRole, interestedJobs]);
+
+  useEffect(() => {
+    const fetchLearnerCourseEnrollments = async () => {
+      try {
+        const response = await fetchCourseEnrollments();
```

Comment:
[curious/clarification] Is this `fetchCourseEnrollments` function returning all enrollments for the learner _across all enterprises_? Should we be referring to only the course enrollments relevant for the currently viewed enterprise customer, instead of all enrollments?

Reply:
@adamstankiewicz  Yes, the `fetchCourseEnrollments` returns a list of all course enrollments for a user. 
In Skills Quiz V1, we utilized this function to fetch the list of all enrollments and for consistency, I replicated the same approach in V2 as well.

Reply:
@mahamakifdar19 While this isn't blocking feedback, I still might recommend verifying with the team / Product whether this should be using all course enrollments vs. just the course enrollments associated with the specific enterprise customer. The Enterprise Learner Portal generally isn't supposed to be aware of B2C enrollments.